wavedump/src/analyze-waveforms

```diff
@@ -90,134 +83,284 @@ def get_bins(x):
-                plt.plot(x,y[:100].T)
-                plt.xlabel("Time (ns)")
-                plt.ylabel("Voltage (V)")
-                plt.title(channel)
-                plt.axvline(x[a])
-                plt.axvline(x[b])
-
-    data['avg_pulse_y'] /= data['avg_pulse_count']
-    data['avg_pulse_x'] = list(map(float,data['avg_pulse_x']))
-    data['avg_pulse_y'] = list(map(float,data['avg_pulse_y']))
-
-    if args.plot:
-        plt.figure()
-
+    f_charge = {}
+    # `average` and `std` just used for testing.
+    # average = {}
+    # std = {}
+    # Only active channel is analyzed, unless it's the empty string, in which case all channels are analyzed.
+    active_ch = args.active
+    for filename in args.filenames:
+        with h5py.File(filename) as f:
+            for channel in f:
+                # All relevant channels, from scope and digitizer, should be in this format: 'ch<channel number>'.
+                if channel == 'settings':
+                    continue
+                if channel.startswith('b'):  # baseline channel, only relevant for testing
+                    continue
+                if active_ch:
+                    if channel != active_ch:
+                        continue
+
+                N = len(f[channel])
+                
+                for i in range(0, N, args.chunks):
+                    x, y = convert_data(f,channel,i,i+args.chunks)
+                    if args.sodium:
+                        a, b = get_window(x,y,left=50,right=300)
+                        y -= np.median(y[:,x < x[0] + 100],axis=-1)[:,np.newaxis]
+                        if channel in charge:
+                            charge[channel] = np.concatenate((charge[channel],integrate(x,y, a, b)))
+                        else:
+                            charge[channel] = integrate(x,y, a, b)
+                    else:
+                        a, b = get_spe_window(x, args.s, args.IT)
+                        # y -= np.median(y[:,x < x[0] + 100])  # per chuck median subtraction
+                        
+                        high_filter_y = high_filter_SPE(x, y)
+                        y -= np.median(y, axis=-1)[:, np.newaxis]  # per event median subtraction
+                        
+                        ################################
+                        # per sample median subtraction
+                        # This method is not good because the SPE signal gets diminished.
+                        ################################
+                        # `s` for samples
+                        # s = y.T
+                        # s_mask = s > -10*iqr(high_filter_y.flatten())
+                        # good_s = [s[i, s_mask[i]] for i in range(len(s))]
+                        # print(f'average number of good samples: {np.mean([len(sub) for sub in good_s])}')
+                        # sample_medians = np.array([np.median(sub) for sub in good_s])
+                        # y -= sample_medians
```

Comment:
For this and all the other methods you define below, I think it's best if we keep them uncommented. There are a couple of ways we could do this. One way is like:

```
# top of file
INTEGRATION_METHOD = 1

if INTEGRATION_METHOD == 1:
    # do per sample median subtraction
elif INTEGRATION_METHOD == 2:
    # do median between `ma` and `mb`
...
```

Alternatively we could have a command line argument:

```python
parser.add_argument("--integration-method",...)
```

but generally we don't want to leave uncommented code because if things get changed in the code and there is now a syntax error or problem it won't get noticed.